ui-speedspacechart/src/components/helpers/drawElements/gridY.ts

```diff
@@ -1,60 +1,31 @@
+import { tickScaleY } from './tickY';
 import type { DrawFunctionParams } from '../../../types/chartTypes';
 import { MARGINS, TICK_TITLE_MARGINS } from '../../const';
 import { clearCanvas, speedRangeValues } from '../../utils';
 
 const { MARGIN_LEFT, MARGIN_RIGHT, MARGIN_TOP, MARGIN_BOTTOM, CURVE_MARGIN_TOP } = MARGINS;
 const { Y_LEFT_VERTICAL, Y_LEFT_HORIZONTAL } = TICK_TITLE_MARGINS;
 
-export const drawGridY = ({ ctx, width, height, store }: DrawFunctionParams) => {
-  const { speeds } = store;
-
-  clearCanvas(ctx, width, height);
-
-  const { maxSpeed } = speedRangeValues(store);
-
-  ctx.strokeStyle = 'rgba(33, 112, 185, 0.25)';
-  ctx.lineWidth = 0.5;
-
-  // horizontal lines based on 10 units of round max speed
-
-  const yPosition = (height - MARGIN_BOTTOM - MARGIN_TOP - CURVE_MARGIN_TOP) / maxSpeed;
-
-  ctx.beginPath();
-  speeds.forEach((_, i) => {
-    if (i >= 0 && i * 10 <= maxSpeed + 10) {
-      ctx.moveTo(MARGIN_LEFT, height - MARGIN_BOTTOM - yPosition * i * 10);
-      ctx.lineTo(width - MARGIN_RIGHT, height - MARGIN_BOTTOM - yPosition * i * 10);
-    }
-  });
-  ctx.closePath();
-  ctx.stroke();
-
-  // prevent overlapping with margin top
-  ctx.clearRect(0, 0, width, MARGIN_TOP);
-  ctx.clearRect(0, height - MARGIN_BOTTOM, width, MARGIN_BOTTOM);
-};
-
 export const drawMajorGridY = ({ ctx, width, height, store }: DrawFunctionParams) => {
-  const { speeds } = store;
-
   clearCanvas(ctx, width, height);
 
   const { maxSpeed } = speedRangeValues(store);
+  const tickScale = tickScaleY(maxSpeed);
+  const nbTicks = Math.ceil(maxSpeed / tickScale);
+  const maxTickSpeed = nbTicks * tickScale;
+  const ratioRoundPositions = maxTickSpeed / maxSpeed;
+  const ticksOffset =
+    ((height - MARGIN_BOTTOM - MARGIN_TOP - CURVE_MARGIN_TOP) * ratioRoundPositions) / nbTicks;
```

Comment:
This block of code (and l.12) is used in both `gridY.ts` and `tickY.ts`. We could create a function in `utils.ts` that returns `tickScale`, `nbTicks` and `ticksOffset`.

Reply:
I was thinking about merging these files into one. What do you think?

Reply:
As both are highly related, I think it should be ok. But as it will become one layer (is that what you meant ?), maybe we should check with @thibautsailly that he doesn't plan later to make the "horizontal lines" an optional option of the speedspace chart.

Reply:
Actually, the plan is to make vertical line optionnal if OPs (steps) are disable. So it makes a lot of sense to have it in the same layer.

Reply:
I merged the two layers. We can always adjust if necessary in the future.

Reply:
and `tickY.ts` -> `gridY.ts`

Reply:
I agree, we have to rename some files to improve visibility and consistency.


> the purpose of having two isolated layers was to facilitate the debug (as for each other layer), but yes it seems ok.

Both layers need the same information. Separating them has the disadvantage of duplicating the code, or requiring a large number of help functions that reduce the readability of the drawing functions.